translate reference/datamodel.po

kisaragi-hiu opened the pull request:
273/804 (33%) -> 307/804 (38%)

Closes #940.

- I added the previous translator to the `Translators:` list.

<details><summary>Preview</summary>
<img src="https://github.com/user-attachments/assets/4d0485c6-e843-423c-a03c-182b4f34dd70" alt="datamodelpreview">
</details>

kisaragi-hiu committed:
translate reference/datamodel.po

mattwang44 commented on reference/datamodel.po:
```diff
@@ -41,10 +47,14 @@ msgid ""
 msgstr ""
+"每個物件都有一個識別性、型別，和數值。物件的\\ *識別性*\\ 在物件建立後永遠不"
+"會改變；您也可以把它想成是該物件在記憶體中的位址。\\ :keyword:`is` 運算子會比"
+"較兩個物件的識別性是否相同；\\ :func:`id` 函式則會回傳代表一個該物件的識別性"
```
```suggestion
"較兩個物件的識別性是否相同；:func:`id` 函式則會回傳代表一個該物件的識別性"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -41,10 +47,14 @@ msgid ""
 #: ../../reference/datamodel.rst:42
 msgid "For CPython, ``id(x)`` is the memory address where ``x`` is stored."
-msgstr ""
+msgstr "在 CPython 當中，\\ ``id(x)`` 就是 ``x`` 所儲存在的記憶體位址。"
```
```suggestion
msgstr "在 CPython 當中，``id(x)`` 就是 ``x`` 所儲存在的記憶體位址。"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -41,10 +47,14 @@ msgid ""
 "identity."
 msgstr ""
+"每個物件都有一個識別性、型別，和數值。物件的\\ *識別性*\\ 在物件建立後永遠不"
+"會改變；您也可以把它想成是該物件在記憶體中的位址。\\ :keyword:`is` 運算子會比"
```
```suggestion
"會改變；您也可以把它想成是該物件在記憶體中的位址。:keyword:`is` 運算子會比"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -54,6 +64,10 @@ msgid ""
 "unchangeable. [#]_"
 msgstr ""
+"一個物件的型別決定了該物件所支援的操作（例如「它有長度嗎？」），也同時定義該"
+"型別的物件能夠擁有的數值。\\ :func:`type` 函式會回傳一個物件的型別（而該型別"
```
```suggestion
"型別的物件能夠擁有的數值。:func:`type` 函式會回傳一個物件的型別（而該型別"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -109,6 +141,11 @@ msgid ""
+"某些物件包含對於「外部」資源的參照，像是開啟的檔案或是視窗。基本上這些資源會"
+"在物件被回收時釋放，但因為垃圾回收不保證會發生，這種物件也會提供明確釋放外部"
+"資源的方式 --- 通常是 :meth:`!close` method。強烈建議各個程式明確關閉這種物"
+"件。\\ :keyword:`try`...\\ :keyword:`finally` 陳述式與 :keyword:`with` 陳述式"
```
```suggestion
"件。:keyword:`try`...\\ :keyword:`finally` 陳述式與 :keyword:`with` 陳述式"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -134,6 +176,13 @@ msgid ""
+"型別影響物件行為的幾乎所有面向。就連物件識別性的重要性某種程度上也受型別影"
+"響：對於不可變的型別，計算新數值的操作可能其實會回傳一個某個相同型別且相同數"
+"值的現存物件的參照；對於可變型別這則不會發生。舉例來說，在進行 ``a = 1; b = "
+"1`` 之後，\\ ``a`` 和 ``b`` 可能會參照同一個物件，也可能不會，取決於所使用的"
+"實作；但在進行 ``c = []; d = []`` 之後，\\ ``c`` 和 ``d`` 則保證會參照兩個不"
+"同、獨特、且新建立的空白串列。（請注意，\\ ``c = d = []`` 則會將同一個物件同"
+"時指派給 ``c`` 和 ``d``\\ 。）"
```
```suggestion
"1`` 之後，``a`` 和 ``b`` 可能會參照同一個物件，也可能不會，取決於所使用的"
"實作；但在進行 ``c = []; d = []`` 之後，``c`` 和 ``d`` 則保證會參照兩個不"
"同、獨特、且新建立的空白串列。（請注意，``c = d = []`` 則會將同一個物件同"
"時指派給 ``c`` 和 ``d``。）"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -193,17 +249,22 @@ msgid ""
 "will raise a :exc:`TypeError` in a future version of Python."
 msgstr ""
+"在預期布林值的情境中計算 :data:`NotImplemented` 的行為已棄用。雖然目前它會計"
+"算為 true，它也會同時發出 :exc:`DeprecationWarning`\\ 。從某個未來 Python 版"
+"本開始這將會引發 :exc:`TypeError`\\ 。"
```
```suggestion
"算為 true，它也會同時發出 :exc:`DeprecationWarning`。從某個未來 Python 版"
"本開始這將會引發 :exc:`TypeError`。"
```

mattwang44 reviewed (commented):
nit: 在（除了括號以外的）全形標點符號旁不用放 `\\ `
（以前認知錯誤所以有些地方還沒修正，但加上去的話 render 也不會有問題）

kisaragi-hiu committed:
Remove extraneous syntax escape as suggested

Co-authored-by: Wei-Hsiang (Matt) Wang <mattwang44@gmail.com>

kisaragi-hiu commented:
移除了不必要的 `\\ `。（太習慣 Org-mode 對全形標點符號的不支援了）

mattwang44 commented on reference/datamodel.po:
```diff
@@ -41,10 +47,14 @@ msgid ""
 "identity."
 msgstr ""
+"每個物件都有一個識別性、型別，和數值。物件的\\ *識別性*\\ 在物件建立後永遠不"
+"會改變；您也可以把它想成是該物件在記憶體中的位址。:keyword:`is` 運算子會比"
```
我們 convention 是統一用「你」不用「您」
```suggestion
"會改變；你也可以把它想成是該物件在記憶體中的位址。:keyword:`is` 運算子會比"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -68,6 +82,13 @@ msgid ""
 "instance, numbers, strings and tuples are immutable, while dictionaries and "
 "lists are mutable."
 msgstr ""
+"某些物件的\\ *數值*\\ 是可變的。數值可變的物件稱作「可變動」(*mutable*)；建立"
+"後數值不可變的物件則稱作「不可變動」(*immutable*)。（不可變的容器物件中如果包"
```
mutable/immutable 現有翻譯為「可變的」、「不可變的」，也許可以修改一下敘述方式，例如 `數值可被改變的物件稱作「可變的」(*mutable*)` 
https://docs.python.org/zh-tw/3/glossary.html#term-mutable

kisaragi-hiu replied:
已改為

    某些物件的\\ *數值*\\ 可被改變，這種物件稱作「可變的」(*mutable*)；建立後數值不能變更的物件則稱作「不可變的」(*immutable*)。

https://github.com/python/python-docs-zh-tw/pull/941/commits/c0a201562946063559faadcf13d15bec5c248914#diff-9241811e9209f72749cbfe61c59fed14e77c739fd9780ddc69f466c4b35100cbL85-R91

mattwang44 commented on reference/datamodel.po:
```diff
@@ -77,6 +98,9 @@ msgid ""
 "objects are collected that are still reachable."
 msgstr ""
+"物件永遠不會被明示的摧毀；但當它們變得不再能夠存取的時候可能會被作為垃圾回"
+"收。每個實作都能延後垃圾回收或是乾脆忽略它 --- 垃圾回收如何進行完全是各個實作"
+"自己的事項，唯一擔保是仍然可存取的物件絕不會被回收。"
```
- nit: 我的感覺是這句話是對實作者說明規範並同時對使用者說明使用行為，但是「擔保」一詞比較像是只針對使用者
- 在[現有的 gc module 翻譯](https://docs.python.org/zh-tw/3/library/gc.html)中，reachable/unreachable 被翻譯成「可達」、「不可達」，但這是沒有經過詳細討論、直接[參考維基百科](https://w.wiki/Auhv)的翻譯，有充分理由就可以推翻。不過不翻譯成「可存取」我覺得也是因為意義上和 `accessible` 有所區別。
```suggestion
"收。每個實作都能延後垃圾回收或是乾脆忽略它 --- 垃圾回收如何進行完全取決於各個"
"實作，只要沒有被回收的物件仍是可達的。"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -121,6 +158,11 @@ msgid ""
+"某些物件包含對於其他物件的參照；這種物件被叫做「容器」。容器的範例有 tuple、"
+"串列與字典。這些參照是容器的數值的一部分。通常當我們提到容器的數值的時候，我"
+"們指的是其中包含的物件的數值，而不是它們的識別性；但當我們提到容器是否可變的"
+"時候，我們指的是單層包含的物件的識別性。因此，如果一個不可變的容器（像一個 "
```
原文指的應該不是「單層」？

kisaragi-hiu replied:
現在試著改成了

```
[...] 但當我們提到容器是否可變的"
"時候，我們指的是直接包含在其中的物件的識別性。因此，如果一個不可變的容器（像"
"一個 tuple）包含對於可變物件的參照，該可變物件被變更時該容器的數值也會跟著變"
"更。"
```

https://github.com/python/python-docs-zh-tw/pull/941/commits/c0a201562946063559faadcf13d15bec5c248914#diff-9241811e9209f72749cbfe61c59fed14e77c739fd9780ddc69f466c4b35100cbR163

mattwang44 commented on reference/datamodel.po:
```diff
@@ -134,6 +176,13 @@ msgid ""
 "different, unique, newly created empty lists. (Note that ``c = d = []`` "
 "assigns the same object to both ``c`` and ``d``.)"
 msgstr ""
+"型別影響物件行為的幾乎所有面向。就連物件識別性的重要性某種程度上也受型別影"
```
nit: 好像比較順？
```suggestion
"型別幾乎影響物件行為的所有面向。就連物件識別性的重要性某種程度上也受型別影"
```

mattwang44 commented on reference/datamodel.po:
```diff
@@ -168,6 +217,9 @@ msgid ""
 msgstr ""
+"這個型別只有一個數值。只有一個物件有這個數值。這個物件由內建名稱 ``None`` 存"
+"取。它用來在許多情況下代表數值不存在，例如沒有明確回傳任何東西的函式就會回傳"
+"這個物件。它的真值是 false。"
```
「真值」感覺會被解讀成 `True`，有更好的翻譯嗎

mattwang44 reviewed (commented):
review part 2

mattwang44 commented:
因應下週 python 正式 release 3.13，main branch 已經切換成 3.13 文件，要再麻煩你 rebase 並切換 target branch 🙏🏽

kisaragi-hiu committed:
Address review

- Mutable and Immutable are 可變 and 不可變 when being defined
- 你 is used here and not 您
- Apply Matt's review for GC:

  垃圾回收如何進行完全是各個實作自己的事項，唯一擔保是仍然可存取的物件絕不會被回收。
  ->
  垃圾回收如何進行完全取決於各個實作，只要沒有被回收的物件仍是可達的。

- Revise away "單層"
- Apply review for 型別幾乎影響物件行為的所有面向

kisaragi-hiu commented:
已 rebase、修改 target 並回應 review; sorry for the delayed response

mattwang44 commented on reference/datamodel.po:
```diff
@@ -138,6 +181,13 @@ msgid ""
+"型別幾乎影響物件行為的所有面向。就連物件識別性的重要性某種程度上也受型別影"
+"響：對於不可變的型別，計算新數值的操作可能其實會回傳一個某個相同型別且相同數"
+"值的現存物件的參照；對於可變型別這則不會發生。舉例來說，在進行 ``a = 1; b = "
+"1`` 之後，``a`` 和 ``b`` 可能會參照同一個物件，也可能不會，取決於所使用的實"
+"作；但在進行 ``c = []; d = []`` 之後，``c`` 和 ``d`` 則保證會參照兩個不同、獨"
+"特、且新建立的空白串列。（請注意，``c = d = []`` 則會將同一個物件同時指派給 "
+"``c`` 和 ``d``。）"
```
原文有所變更，中間有一段是新的敘述（我先翻了一版）
```suggestion
"1`` 之後，*a* 和 *b* 可能會參照同一個物件，也可能不會，取決於所使用的實"
"作。"
"這是因為 :class:`int` 為不可變型別，所以對 ``1`` 的參照可以被複用，此行為"
"取決於所使用的實作，所以不應該依賴於此，但會是個使用物件識別性測試時需要注意的事情；"
"在進行 ``c = []; d = []`` 之後，*c* 和 *d* 則保證會參照兩個不同、獨"
"特、且新建立的空白串列。（請注意，``e = f = []`` 則會將同一個物件同時指派給 "
"*e* 和 *f*。）"
```

kisaragi-hiu committed:
Translate missed source text update

mattwang44 reviewed (approved):
LGTM!
